### PR TITLE
Upgrade robolectric and target api 29.

### PR DESCRIPTION
**Pull request by vkryachko**
This change mostly addresses: https://github.com/robolectric/robolectric/issues/5456

### CODE REVIEW

File: firebase-database/src/main/java/com/google/firebase/database/tubesock/WebSocket.java [hunk 1]
```diff
@@ -14,7 +14,6 @@
 
 package com.google.firebase.database.tubesock;
 
-import android.net.SSLCertificateSocketFactory;
```
Comment (ashwinraghav) [inline firebase-database/src/main/java/com/google/firebase/database/tubesock/WebSocket.java:17]: Intentional?

File: firebase-firestore/src/main/java/com/google/firebase/firestore/remote/AndroidConnectivityMonitor.java [hunk 2]
```diff
@@ -124,10 +123,11 @@ public void onLost(Network network) {
     private boolean isConnected = false;
 
     @Override
+    @SuppressWarnings("deprecation")
     public void onReceive(Context context, Intent intent) {
       ConnectivityManager conn =
           (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
-      NetworkInfo networkInfo = conn.getActiveNetworkInfo();
+      android.net.NetworkInfo networkInfo = conn.getActiveNetworkInfo();
```
Comment (ashwinraghav) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/remote/AndroidConnectivityMonitor.java:130]: Intentional?

File: firebase-storage/src/main/java/com/google/firebase/storage/network/NetworkRequest.java [hunk 3]
```diff
@@ -235,7 +235,7 @@ public void performRequestStart(String token) {
     }
     ConnectivityManager connMgr =
         (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
-    NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
+    android.net.NetworkInfo networkInfo = connMgr.getActiveNetworkInfo();
```
Comment (ashwinraghav) [inline firebase-storage/src/main/java/com/google/firebase/storage/network/NetworkRequest.java:238]: Ditto

File: buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/ClosureUtil.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins;
+
+import groovy.lang.Closure;
+import java.util.function.Consumer;
+
+public final class ClosureUtil {
```
Comment (ashwinraghav) [inline buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/ClosureUtil.java:20]: Can You help understand what this does?

File: firebase-database/src/main/java/com/google/firebase/database/tubesock/WebSocket.java (hunk 1 above)
Comment (vkryachko) [reply firebase-database/src/main/java/com/google/firebase/database/tubesock/WebSocket.java:17]: Avoiding deprecation warnings, which are configured as hard errors in these projects

File: buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/ClosureUtil.java (hunk 4 above)
Comment (vkryachko) [reply buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/ClosureUtil.java:20]: this is to express the following in java(this api is not java-friendly):
```groovy
android.testOptions.unitTests.all {
 // this is a groovy closure that is hard to implement in java, hence this util class
}
```

**Pull request merged**